backend/server-connection/src/main/java/org/sonarsource/sonarlint/core/serverconnection/storage/ServerInfoStorage.java

```diff
@@ -81,6 +81,10 @@ private static StoredServerInfo adapt(Sonarlint.ServerInfo serverInfo) {
       }
       globalSettings.put(ServerSettings.EARLY_ACCESS_MISRA_ENABLED, Boolean.toString(serverInfo.getMisraEarlyAccessRulesEnabled()));
     }
+    if (globalSettings.containsKey(ServerSettings.EARLY_ACCESS_MISRA_ENABLED)) {
+      globalSettings = new HashMap<>(globalSettings);
+      globalSettings.put(ServerSettings.MISRA_COMPLIANCE_ENABLED, globalSettings.get(ServerSettings.EARLY_ACCESS_MISRA_ENABLED));
+    }
```

Comment:
I presume this will serve only until the next synchronization occurs, when the new setting will be downloaded? Maybe you could leave a comment, it was not obvious what is the purpose of this if

Reply:
Ah yes, because the embedded CFamily analyzer is used even in connected mode. All good!